pyam/core.py

```diff
@@ -518,35 +518,73 @@ def validate(self, criteria={}, exclude_on_fail=False):
 
             return df
 
-    def rename(self, mapping, inplace=False):
+    def rename(self, mapping=None, inplace=False, append=False, **kwargs):
         """Rename and aggregate column entries using `groupby.sum()` on values.
         When renaming models or scenarios, the uniqueness of the index must be
         maintained, and the function will raise an error otherwise.
 
+        Renaming is only applied to any data where a filter matches for all
+        columns given in `mapping`. Renaming can only be applied to the `model`
+        and `scenario` columns or to other data columns simultaneously.
+
         Parameters
         ----------
-        mapping: dict
-            for each column where entries should be renamed, provide current
-            name and target name
-            {<column name>: {<current_name_1>: <target_name_1>,
-                             <current_name_2>: <target_name_2>}}
+        mapping: dict or kwargs
+            mapping of column name to rename-dictionary of that column
+            >> {<column_name>: {<current_name_1>: <target_name_1>,
+            >>                  <current_name_2>: <target_name_2>}}
+            or kwargs as `column_name={<current_name_1>: <target_name_1>, ...}`
         inplace: bool, default False
             if True, do operation inplace and return None
+        append: bool, default False
+            if True, append renamed timeseries to IamDataFrame
         """
+        # combine `mapping` arg and mapping kwargs, ensure no rename conflicts
+        mapping = mapping or {}
```

Comment:
should we add a deprecation warning for mapping?

Reply:
I went back and forth about that a few times, but in the end, I thought that it would be better to keep both options permanently for mimicking the multiple options in [pd.DataFrame.rename()](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.rename.html).